cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedTimeScheduleStore.java

```diff
@@ -173,7 +202,7 @@ private void executeDelete(final TriggerKey triggerKey) {
         .execute(new TransactionExecutor.Subroutine() {
           @Override
           public void apply() throws Exception {
-            if (versionLessTriggerKey != null && readTrigger(versionLessTriggerKey) != null) {
+            if (!isUpgradeComplete() && versionLessTriggerKey != null && readTrigger(versionLessTriggerKey) != null) {
```

Comment:
only need to call `removeAppVersion(triggerKey)` if `!isUpgradeComplete()`, so that no need of `!isUpgradeComplete() &&` here